internal/stacksapi/client.go

```diff
@@ -0,0 +1,263 @@
+package stacksapi
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log/slog"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/buildkite/agent-stack-k8s/v2/internal/version"
+	"github.com/buildkite/roko"
+)
+
+type Client struct {
+	baseURL    *url.URL
+	userAgent  string
+	authHeader string
+
+	logger      *slog.Logger
```

Comment:
Oooh, is the thinking we'll centralise on `slog`? 😀

Reply:
god, i sure hope so.